src/pages/home/report/EmojiSkinToneList.js

```diff
@@ -26,76 +27,88 @@ const propTypes = {
+     * @param {object} skinToneEmoji
+     */
+    updateSelectedSkinTone(skinToneEmoji) {
+        this.setState(prev => ({isSkinToneListVisible: !prev.isSkinToneListVisible}));
+        this.props.setPreferredSkinTone(skinToneEmoji.skinTone);
+    }
 
-    );
-};
+    render() {
+        const selectedEmoji = getSkinToneEmojiFromIndex(this.props.preferredSkinTone);
+        return (
+            <View style={[styles.flexRow, styles.p1]}>
+                {
+                    !this.state.isSkinToneListVisible && (
+                        <Pressable
+                            onPress={
+                                () => this.setState(prev => ({isSkinToneListVisible: !prev.isSkinToneListVisible}))
+                            }
+                            style={[
+                                styles.pv1,
+                                styles.flex1,
+                                styles.flexRow,
+                                styles.alignSelfCenter,
+                                styles.justifyContentStart,
+                            ]}
+                        >
+                            <Text style={[styles.emojiText, this.props.emojiSize]}>
+                                {`${selectedEmoji.code}\uFE0F`}
+                            </Text>
+                            <Text style={[styles.emojiHeaderStyle]}>
+                                {this.props.translate('emojiPicker.skinTonePickerLabel')}
+                            </Text>
+                        </Pressable>
+                    )
+                }
+                {
+                    this.state.isSkinToneListVisible && (
+                        <View>
+                            <View style={[styles.flexRow]}>
+                                {
+                                    skinTones.map(skinToneEmoji => (
+                                        <EmojiPickerMenuItem
+                                            onPress={() => this.updateSelectedSkinTone(skinToneEmoji)}
+                                            onHover={() => this.setState({highlightedIndex: skinToneEmoji.skinTone})}
+                                            key={skinToneEmoji.code}
+                                            emojiItemStyle={[styles.emojiSkinToneItem, styles.emojiSkinToneItem]}
+                                            emoji={`${skinToneEmoji.code}\uFE0F`}
+                                            isHighlighted={skinToneEmoji.skinTone === this.state.highlightedIndex}
+                                            emojiSize={this.props.emojiSize}
+                                        />
+                                    ))
+                                }
+                            </View>
+                        </View>
+                    )
+                }
+            </View>
+        );
+    }
+}
 
 EmojiSkinToneList.propTypes = propTypes;
-EmojiSkinToneList.defaultProps = defaultProps;
-
+EmojiSkinToneList.defaultProps = {};
```

Comment:
We can just get rid of this line right? Since all props are required?

Reply:
Updated